components/form/TaskForm.tsx

```diff
@@ -1,30 +1,31 @@
 import { useUpdateTasks } from '@/lib/hooks/useTasks';
 import { Project, Task } from '@/lib/types/apiTypes';
-import { FormControl, FormLabel, Input, FormErrorMessage, ModalFooter, Button, Box } from '@chakra-ui/react';
+import { FormBase } from '@/lib/types/forms';
+import { FormControl, FormLabel, Input, FormErrorMessage, Button, Box } from '@chakra-ui/react';
 import React, { useState } from 'react';
 import ErrorAlert from '../common/ErrorAlert';
 
 type TaskFields = Partial<Task> & { project: Project };
 
-const validateTaskFields = (fields: TaskFields): Task => {
-    const { name } = fields;
+const validateTaskFields = (fields: TaskFields, projectId: number): Task => {
+    const { name, project } = fields;
     if (name) {
         return {
             ...fields,
+            project: { ...project, id: projectId },
```

Comment:
This is important BTW, a missing `projectId` could result in some really nasty hard-to-find bugs